src/python/pants/backend/python/goals/setup_py.py

```diff
@@ -200,7 +200,14 @@ def __init__(
             raise ValueError(f"Missing a `version` kwarg in the `provides` field for {address}.")
 
         if not _allow_banned_keys:
-            for arg in {"data_files", "package_dir", "package_data", "packages"}:
+            for arg in {
+                "data_files",
+                "namespace_packages",
+                "package_dir",
+                "package_data",
+                "packages",
+                "install_requires",
+            }:
```

Comment:
Alternatively, we could allow users to set some of these and have us extend the values, rather than overwriting like we were doing.

I think I err on the side of banning, though. As a user, if I set `namespace_packages=['foo']`, I think I would be confused if Pants ended up adding two additional values to it. It's not clear that Pants auto-magically sets these 6 values, but it doesn't touch any other kwargs. 

If we get user feedback that they really need to set these, we can add a safety valve. (Actually, that safety valve already exists. They can use a custom `SetupKwargs` implementation and in the constructor to `SetupKwargs`, set the arg `_allow_banned_keys=True`.

Reply:
Would that work, since we overwrite? Wouldn't we have to check if the value exists and only write it if not?

Reply:
> Would that work, since we overwrite? Wouldn't we have to check if the value exists and only write it if not?

Ah, true. I can fix that, though, if you think it's a good idea?
 
That would mean that by default, you can't set those values because it would be too confusing. But if you really want a safety valve, you have it.

Reply:
Yes, exactly. With a comment explaining that :)